### ISSUE TITLE
obs-ndi.so causes OBS to fail to start, even though NDI appears to have been initialized successfully.

### ISSUE THREAD

**Issue by qubitrenegade** [obs-ndi/obs-ndi#1006]
Hello,

I have just built obs-ndi for a Raspberry Pi 5.

I added the steps I went through to the wiki: https://github.com/obs-ndi/obs-ndi/wiki/3.-Development#raspbian

However, when I attempt to start OBS, I get the following log info, and then obs doesn't start:

```
2024-04-26 01:26:32,019 started command 'obs' with pid 72987
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
debug: Found portal inhibitor
debug: Attempted path: share/obs/obs-studio/locale/en-US.ini
debug: Attempted path: /usr/share/obs/obs-studio/locale/en-US.ini
debug: Attempted path: share/obs/obs-studio/locale.ini
debug: Attempted path: /usr/share/obs/obs-studio/locale.ini
debug: Attempted path: share/obs/obs-studio/locale/en-GB.ini
debug: Attempted path: /usr/share/obs/obs-studio/locale/en-GB.ini
info: Using preferred locale 'en-GB'
debug: Attempted path: share/obs/obs-studio/themes/Yami.qss
debug: Attempted path: /usr/share/obs/obs-studio/themes/Yami.qss
info: Using EGL/X11
info: Physical Cores: 4, Logical Cores: 4
info: Physical Memory: 8052MB Total, 4310MB Free
info: Kernel Version: Linux 6.6.20+rpt-rpi-2712
info: Distribution: "Debian GNU/Linux" "12"
info: Session Type: x11
info: Window System: X11.0, Vendor: The X.Org Foundation, Version: 1.21.1
info: Qt Version: 5.15.8 (runtime), 5.15.8 (compiled)
info: Portable mode: false
QMetaObject::connectSlotsByName: No matching signal for on_tbar_position_valueChanged(int)
info: OBS 29.0.2.1-1+b1 (linux)
info: ---------------------------------
info: ---------------------------------
info: audio settings reset:
        samples per sec: 48000
        speakers:        2
        max buffering:   960 milliseconds
        buffering type:  dynamically increasing
info: ---------------------------------
info: Initializing OpenGL...
info: Loading up OpenGL on adapter Mesa llvmpipe (LLVM 15.0.6, 128 bits)
info: OpenGL loaded successfully, version 4.5 (Core Profile) Mesa 23.2.1-1~bpo12+rpt3, shading language 4.50
info: ---------------------------------
info: video settings reset:
        base resolution:   1920x1080
        output resolution: 1280x720
        downscale filter:  Bicubic
        fps:               30/1
        format:            NV12
        YUV mode:          Rec. 709/Partial
info: NV12 texture support not available
info: P010 texture support not available
info: Audio monitoring device:
        name: Default
        id: default
info: ---------------------------------
warning: Failed to load 'en-US' text for module: 'decklink-captions.so'
warning: Failed to load 'en-US' text for module: 'decklink-output-ui.so'
libDeckLinkAPI.so: cannot open shared object file: No such file or directory
warning: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed
warning: Failed to initialize module 'decklink.so'
warning: Failed to load 'en-GB' text for module: 'frontend-tools.so'
warning: Failed to load 'en-GB' text for module: 'linux-alsa.so'
warning: Failed to load 'en-GB' text for module: 'linux-capture.so'
warning: Failed to load 'en-GB' text for module: 'linux-jack.so'
warning: Failed to load 'en-GB' text for module: 'linux-pipewire.so'
info: [pipewire] No captures available
warning: Failed to load 'en-GB' text for module: 'linux-pulseaudio.so'
warning: v4l2loopback not installed, virtual camera disabled
error: VAAPI: Failed to initialize display in vaapi_device_h264_supported
info: FFmpeg VAAPI H264 encoding not supported
warning: Failed to load 'en-GB' text for module: 'obs-ndi.so'
info: [obs-ndi] obs_module_load: you can haz obs-ndi (Version 4.13.1)
info: [obs-ndi] obs_module_load: Qt Version: 6.4.2 (runtime), 6.4.2 (compiled)
info: [obs-ndi] load_ndilib: Trying '/usr/lib/libndi.so.5'
info: [obs-ndi] load_ndilib: Trying '/usr/local/lib/libndi.so.5'
info: [obs-ndi] load_ndilib: Found NDI library at '/usr/local/lib/libndi.so.5'
info: [obs-ndi] load_ndilib: NDI runtime loaded successfully
info: [obs-ndi] load_ndilib: NDIlib_v5_load found
info: [obs-ndi] obs_module_load: NDI library initialized successfully ('NDI SDK LINUX 12:52:37 Feb 16 2024 5.6.1')
info: [obs-ndi] preview_output_init('OBS Preview')
info: [obs-ndi] +ndi_output_create('OBS Preview'...)
info: [obs-ndi] ndi_output_update('OBS Preview'...)
info: [obs-ndi] -ndi_output_create('OBS Preview'...)
QWidget: Must construct a QApplication before a QWidget
Aborted
```

If I `rm /usr/lib/aarch64-linux-gnu/obs-plugins/obs-ndi.so`, then OBS starts no problem (but then I have no ndi).

So I'm stumped.

**Comment by tt2468** [comment]
Your obs-ndi is compiled for Qt 6, however your OBS is compiled for Qt 5. You must use the same Qt (and recommended same minor) version for OBS plugins as with OBS itself, otherwise undefined behavior will occur. Also, OBS is running on llvmpipe (software rendering), so I would expect VERY bad performance.

Interestingly, obs-ndi seems to be reporting it's running with Qt 6.4.2, so I guess maybe being linked to Qt 6, it's actually loading the Qt6 binaries for itself.

**Comment by qubitrenegade** [comment]
ah ha!

Thanks @tt2468!

Basically, I just needed to set `QT_VERSION=5` when running the build script.  I updated my notes in the wiki, but the full command was:

QT_VERSION=5 .github/scripts/build-linux --target linux-aarch64

Unfortunately, it seems the performance is pretty poor...  there seems to be about 30s of lag between what OBS on the PI sees and what is viewed via NDI stream...  also a lot of dropped frames on the Pi side of things.

Unfortunately, it seems the forum wisdom of starting obs with `MESA_GL_VERSION_OVERRIDE=3.3 obs` does not stop OBS from using llvmpipe.

```
info: Loading up OpenGL on adapter Mesa llvmpipe (LLVM 15.0.6, 128 bits)
info: OpenGL loaded successfully, version 3.3 (Core Profile) Mesa 23.2.1-1~bpo12+rpt3, shading language 4.50
```

But that is not an issue for the `obs-ndi` folks, so I'll go ahead an mark this closed!

Thanks again for the suggestion @tt2468!!
